Missing classes detected while running R8

If we are using 
minifyEnabled true
shrinkResources true


ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /Users/vikashkumar/FlutterProjects/Test/minglemate/build/app/outputs/mapping/release/missing_rules.txt.
ERROR: R8: Missing class org.tensorflow.lite.gpu.GpuDelegateFactory$Options (referenced from: void org.tensorflow.lite.gpu.GpuDelegate.<init>() and 1 other context)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 42s
Running Gradle task 'assembleRelease'...                           42.7s
Gradle task assembleRelease failed with exit code 1


missing_rules.txt file content


# Please add these rules to your existing keep rules in order to suppress warnings.
# This is generated automatically by the Android Gradle plugin.
-dontwarn org.tensorflow.lite.gpu.GpuDelegateFactory$Options

--- a/README.md
+++ b/README.md
@@ -96,12 +96,36 @@ If there are issues related to the library on iOS and it doesn't work, check the
 
 ### Android
 
-This package uses the `tflite_flutter` package, so the Android `minSdkVersion` must be set to 26 or higher. Check the following setting in the `android/app/build.gradle` file:
+#### 1.	Min SDK Version:
+
+This package uses the `tflite_flutter` package, so the Android `minSdkVersion` must be set to 26 or higher. Check and update the following in your `android/app/build.gradle` file:
 
 ```gradle
 android {
     defaultConfig {
         minSdkVersion 26
+    }
+}
+```
+
+#### 2.	ProGuard / R8 Configuration:
+
+If you encounter issues during release builds due to R8 (ProGuard), ensure that the following rules are added to your `android/app/proguard-rules.pro` file:
+
+```pro
+# TensorFlow Lite GPU Delegate
+-keep class org.tensorflow.lite.gpu.GpuDelegateFactory$Options { *; }
+-dontwarn org.tensorflow.lite.gpu.GpuDelegateFactory$Options
+```
+
+Also, verify that ProGuard or R8 is enabled in your `android/app/build.gradle` file:
+
+```gradle
+buildTypes {
+    release {
+        minifyEnabled true
+        shrinkResources true
+        proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
     }
 }
 ```

